domain/src/main/kotlin/no/nav/su/se/bakover/domain/grunnlag/Grunnlag.kt

```diff
@@ -189,6 +236,55 @@ sealed class Grunnlag {
             fun List<Bosituasjon>.oppdaterBosituasjonsperiode(oppdatertPeriode: Periode): List<Bosituasjon> {
                 return this.map { it.oppdaterBosituasjonsperiode(oppdatertPeriode) }
             }
+
+            fun List<Bosituasjon>.slåSammenPeriodeOgBosituasjon(): List<Bosituasjon> {
+                return this.sortedBy { it.periode.fraOgMed }
+                    .fold(mutableListOf<MutableList<Bosituasjon>>()) { acc, bosituasjon ->
+                        if (acc.isEmpty()) {
+                            acc.add(mutableListOf(bosituasjon))
+                        } else if (acc.last().sisteBosituasjonsgrunnlagErLikOgTilstøtende(bosituasjon)) {
+                            acc.last().add(bosituasjon)
+                        } else {
+                            acc.add(mutableListOf(bosituasjon))
+                        }
+                        acc
+                    }.map {
+                        val periode: Periode = Periode.create(
+                            fraOgMed = it.minOf { it.periode.fraOgMed },
```

Comment:
Denne finnes vel som en extension function i Periode? kunne vi it.map{it.periode}.TheFunction